src/basic_recipes/text.jl

```diff
@@ -65,7 +65,7 @@ function plot!(plot::Text)
     plot
 end
 
-function _get_glyphcollection_and_linesegments(string::String, index, ts, f, al, rot, jus, lh, col, scol, swi, www)
+function _get_glyphcollection_and_linesegments(string::AbstractString, index, ts, f, al, rot, jus, lh, col, scol, swi, www)
     gc = layout_text(string, ts, f, al, rot, jus, lh, col, scol, swi, www)
```

Comment:
```suggestion
function _get_glyphcollection_and_linesegments(str::AbstractString, index, ts, f, al, rot, jus, lh, col, scol, swi, www)
    gc = layout_text(string(str), ts, f, al, rot, jus, lh, col, scol, swi, www)
```

Probably best to be on the safe side here.  Not sure if this would error with substrings.

Reply:
`string(str)` makes a copy of `str` if it is a `SubString` and returns a `String`, so this should work. In case one wants to update the text by changing the view of the `SubString` one should already pass in an `Observable`, I guess.

However, `layout_text` is already typed with  `AbstractString`. If it expects only a `String` then we should update the signature there too, or? Alternatively, we could move the `string` into `layout_text`. `string(str)` is a no-op if `str isa String`.

Reply:
I mean realistically if one has an AbstractString that does not behave like String/Substring for all usual operations then `layout_text` will just not work. Or like in the case of LaTeXString needs its own layout function anyway. So conversion to string is probably fine.